apache/superset
# fix(reports): Owners no longer showing undefined in reports

### SUMMARY

"undefined undefined" was showing up in the owners list of a report every time it was edited. This was resolved by setting the `owners` data to match the state in `alert` (straight from redux state) instead of `resource` (coming from the alert fetch logic). Since `resource` is manipulated in the update, sometimes `owners` will be a list of owner IDs with no other data. This caused the label to return "undefined" for `first_name` and `last_name` in the `owners` object when it's updated. `alert`'s state, however, consistently has the entire `owners` object. It now updates appropriately when updating the owners on a report.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

#### BEFORE
![d08ac168-13e7-44b3-9fab-52994186c68b-hs_file_upload-report_bug (1)](https://user-images.githubusercontent.com/55605634/138965257-a43ea3f1-4fed-4a96-b353-11197835bc42.png)

#### AFTER
![definedOwners](https://user-images.githubusercontent.com/55605634/138965624-c29855c2-60d7-46e7-9c6b-8247751b44e1.gif)


### TESTING INSTRUCTIONS

- Create and save a report
- Edit the report and add another owner
- Save changes
- Go back and edit report again
- Observe that all owners are displaying correctly, no instances of "undefined undefined"

### ADDITIONAL INFORMATION
